binary-quantization-mistral/bq-mistral-embed.md

```diff
@@ -0,0 +1,537 @@
+collection_info.vectors_count
+```
+
+### Configuring Indexing
+
+In this segment, the procedure focuses on enhancing the search operation within a certain collection by adjusting its indexing settings through the `update_collection` method. Specifically, it updates the configuration of a collection (identified by `collection_name`) to adjust its `indexing_threshold`. The `indexing_threshold` parameter is crucial as it determines the number of vectors that should accumulate before the indexing process is triggered. By setting the `indexing_threshold` to 20,000, it implies that the collection will start indexing its contents once the count of stored vectors reaches 20,000. This operation is facilitated with the assistance of `models.OptimizersConfigDiff`, a part of the Qdrant client library designed to fine-tune the performance and efficiency of vector searches.
+
+The presented code doesn't produce a visible output (`['']`), suggesting that the operation completed without returning any direct data. However, implicitly, this operation modifies the configuration of the collection with respect to its indexing behavior, defining how and when the vectors stored within the collection are indexed. This step is significant as proper indexing is key to achieving faster and more efficient vector searches, particularly in large-scale datasets where search performance can be critical. By tailoring the `indexing_threshold`, users can balance between the immediacy of indexing and resource utilization, optimizing the collection's performance for specific use cases or constraints.
+
+```python
+client.update_collection(
+    collection_name=f"{collection_name}",
+    optimizer_config=models.OptimizersConfigDiff(indexing_threshold=20000),
+)
+```
+
+### Advanced Search Options with Qdrant
+
+Qdrant is a vector search engine designed for efficient similarity search in large datasets. It is particularly useful for machine learning engineers and data scientists who need to manage and query embeddings or high-dimensional vectors. One of the key features of Qdrant is its ability to perform both approximate and exact searches, depending on the requirements of the task.
+
+### Performing a Search in Qdrant
+
+Using Qdrant to conduct a search involves specifying both the collection to search within and the query vector. The collection contains the indexed vectors, while the query vector represents the item you are searching for. The goal is to find the most similar vectors to the query vector within the collection.
+
+The following code snippet demonstrates how to perform a search in Qdrant:
+
+```python
+client.search(
+    collection_name=f"{collection_name}",
+    query_vector=points[32].vector,
+    search_params=models.SearchParams(
+        quantization=models.QuantizationSearchParams(
+            ignore=False,
+            rescore=False,
+            oversampling=2.0,
+        ),
+        exact=True,
+    ),
+    limit=5,
+)
```

Comment:
magic indexation `points[32]` :(

quantization:
ignore = False - do not ignore quantized vectors, use them
rescore = False - ignore original vectors, do not rescore search result
oversampling = 2.0 - during search look for `2x * limit` quantized vectors

limit = 5 

so the search is done in the following way:
find 10 nearest neighbors based on quantized vectors, then just drop 5 of them without re-scoring

what I wanted to say:
`oversampling` does not make sense without `rescore`